lsp/protocol/extensions.go

```diff
@@ -0,0 +1,151 @@
+package protocol
```

Comment:
I am hopeful that we can move to https://github.com/go-language-server/protocol but I think that's an `important-long-term` type issue. Do we agree? I can writeup the issue later.

Reply:
Definitely something I want to investigate. Any way we can make it less brittle is a win in my book